cranelift/simplejit/src/backend.rs

```diff
@@ -129,9 +131,13 @@ pub struct SimpleJITModule {
     libcall_names: Box<dyn Fn(ir::LibCall) -> String>,
     memory: MemoryHandle,
     declarations: ModuleDeclarations,
+    function_got_entries: SecondaryMap<FuncId, Option<NonNull<*const u8>>>,
+    function_plt_entries: SecondaryMap<FuncId, Option<NonNull<[u8; 16]>>>,
```

Comment:
I'm not clear on how 16 bytes was arrived at, and how platform specific this is - is it due to 16-byte alignment on x86_64? does some other sequence of instructions have to fit in here besides the plt_val at line 179?

Reply:
In the future I may extend it to be more like the lazy symbol binding used by ELF. In that case the GOT entry will initially point just after the jump instruction in the PLT. After this jump instruction it pushes a specific value to the stack depending on which PLT entry it is and then it jumps to a function common between all entries in a module. This function then resolves the symbol. This avoids having to create a new wrapper function for each lazily compiled function.

Reply:
That makes sense. a comment to this effect would be appreciated :)